aspnetcore/fundamentals/minimal-apis/handle-errrors.md

```diff
@@ -0,0 +1,165 @@
+
+For more information on configuring middleware see, [Middleware in Minimal API apps](/aspnet/core/fundamentals/minimal-apis/middleware).
+
+Using the preceding Minimal API app, when the `Developer Exception Page` detects an unhandled exception, it generates a default plain-text response similar to the following example:
+
+```console
+HTTP/1.1 500 Internal Server Error
+Content-Type: text/plain; charset=utf-8
+Date: Thu, 27 Oct 2022 18:00:59 GMT
+Server: Kestrel
+Transfer-Encoding: chunked
+ 
+    System.InvalidOperationException: Sample Exception
+    at Program.<>c.<<Main>$>b__0_1() in ....:line 17
+    at lambda_method2(Closure, Object, HttpContext)
+    at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext)
+    --- End of stack trace from previous location ---
+    at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
+HEADERS
+=======
+Accept: */*
+Connection: keep-alive
+Host: localhost:5239
+Accept-Encoding: gzip, deflate, br
+```
+
+> [!WARNING]
+> Don't enable the Developer Exception Page **unless the app is running in the Development environment**. Don't share detailed exception information publicly when the app runs in production. For more information on configuring environments, see <xref:fundamentals/environments>.
+
+### Exception handler
+
+In non-development environments, use [Exception Handler Middleware](xref:fundamentals/error-handling#exception-handler-page) to produce an error payload. To configure the `Exception Handler Middleware` in the preceding Minimal API app, call <xref:Microsoft.AspNetCore.Builder.ExceptionHandlerExtensions.UseExceptionHandler%2A>. 
+
+For example, the following code change the app to respond an [RFC 7807](https://tools.ietf.org/html/rfc7807)-compliant payload to the client. For more information, see [Problem Details](#problem-details) section.
+
+
+``` csharp
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.UseExceptionHandler(exceptionHandlerApp 
+    => exceptionHandlerApp.Run(async context 
+        => await Results.Problem()
+                     .ExecuteAsync(context)));
+
+app.Map("/exception", () 
+    => { throw new InvalidOperationException("Sample Exception"); });
+
+app.Run();
+```
+
+## Client and Server error responses
+
+Consider the following Minimal API app.
+
+``` csharp
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.Map("/users/{id:int}", (int id) 
+    => id <= 0 ? Results.BadRequest() : Results.Ok(new User(id)) );
+
+app.Run();
+
+public record User(int Id);
+```
+
+The `/users` endpoint produces `200 OK` with a `json` representation of `User` when `id` greater than `0`, or, `400 BAD REQUEST` status code without a response body. For more information about creating response, see [Create responses in Minimal API apps](/aspnet/core/fundamentals/minimal-apis/responses).
```

Comment:
```suggestion
The `/users` endpoint produces `200 OK` with a `json` representation of `User` when `id` is greater than `0`, otherwise a `400 BAD REQUEST` status code without a response body. For more information about creating response, see [Create responses in Minimal API apps](/aspnet/core/fundamentals/minimal-apis/responses).
```